hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/TokenContextImpl.java

```diff
@@ -84,6 +84,11 @@ public <T> T userTransactionRecordBuilder(@NonNull Class<T> recordBuilderClass)
         return castRecordBuilder(recordListBuilder.userTransactionRecordBuilder(), recordBuilderClass);
     }
 
+    @Override
+    public boolean hasChildRecords() {
+        return !recordListBuilder.childRecordBuilders().isEmpty();
```

Comment:
Should this include precedingChildRecords also ?

Reply:
Agreed Neeha, that would be more future-proof.